source/class/qx/io/graphql/protocol/Request.js

```diff
@@ -40,18 +40,47 @@ qx.Class.define("qx.io.graphql.protocol.Request",{
     variables : {
       check: "qx.core.Object",
       nullable: true,
-      event: "changeVariables"
+      event: "changeVariables",
+      transform: "_transformVariables",
+      validate: "_validateVariables"
     }
   },
 
   members: {
+    // overriden
+    toString() {
+      return qx.lang.Json.stringify(this.toObject(), this._jsonReplacer);
+    },
+
+    _jsonReplacer: function(key, value) {
+      // Special case. If the variables key is an object, return it's JSON
+      // representation, otherwise remove that key from the final JSON
+      if (key === "variables") {
+        return value !== null ? qx.util.Serializer.toJson(value) : undefined;
+      }
+      // everything else s returned as it is
+      return value;
+    },
+
+
     /**
-     * Set the variables from a native javascript object, which will be
-     * marshaled into a qooxdoo object which can be used in databinding
-     * @param {Object} map
+     * Transforms the variables object to a qooxdoo model. Called automaticaly
+     * when the variables property is set.
      */
-    marshalVariables(map) {
```

Comment:
Loosing a public method ... does no one use that?

Reply:
This is ok since we're technically in alpha. So far, it's only used in the tests, I think.